map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,201 @@
+        SimpleLineSymbol.Style.SOLID, 0xFF00FF00, 2.0f);
+      SimpleRenderer locationHistoryLineRenderer = new SimpleRenderer(locationLineSymbol);
+      locationHistoryLineOverlay.setRenderer(locationHistoryLineRenderer);
+
+      // add the graphics overlays to the map view
+      mapView.getGraphicsOverlays().addAll(Arrays.asList(locationHistoryOverlay, locationHistoryLineOverlay));
+
+      // create a polyline builder to connect the location points
+      PolylineBuilder polylineBuilder = new PolylineBuilder(SpatialReference.create(3857));
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+        StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline routePolyline = (Polyline) Polyline.fromJson(polylineData,
+        SpatialReference.create(3857));
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        routePolyline, new SimulationParameters(Calendar.getInstance(), 30.0, 0.0, 0.0));
+
+      // start the simulated location data source
+      simulatedLocationDataSource.startAsync();
+      simulatedLocationDataSource.addLocationChangedListener(locationChangedEvent -> {
+        // if location tracking is turned off, do not add to the polyline
+        if (!isTrackingEnabled) {
+          return;
+        }
+        // get the position as a point from locationChangedEvent
+        Point position = locationChangedEvent.getLocation().getPosition();
+
+        // add the new point to the polyline
+        polylineBuilder.addPoint(position);
+
+        // add the new point to the graphics overlay
+        locationHistoryOverlay.getGraphics().add(new Graphic(position));
+
+        // reset the old polyline connecting the points
+        locationHistoryLineOverlay.getGraphics().clear();
+
+        // add the updated polyline to the graphics overlay
+        locationHistoryLineOverlay.getGraphics().add(new Graphic(polylineBuilder.toGeometry()));
+      });
+
+      // configure the map view's location display to follow the simulated location data source
+      mapView.getLocationDisplay().setLocationDataSource(simulatedLocationDataSource);
+      mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      mapView.getLocationDisplay().setInitialZoomScale(7000);
+
+      trackingButton.setOnAction(event -> {
+        // if the user has panned away from the location display, turn it on again
+        if (mapView.getLocationDisplay().getAutoPanMode() == LocationDisplay.AutoPanMode.OFF) {
+          mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
```

Comment:
Instead of repeatedly querying the mapView's LocationDisplay, we can save it to a variable. Although it adds a line of code, it makes the other lines a bit clearer:

```suggestion
	  LocationDisplay locationDisplay = mapView.getLocationDisplay();
      locationDisplay.setLocationDataSource(simulatedLocationDataSource);
      locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
      locationDisplay.setInitialZoomScale(7000);

      trackingButton.setOnAction(event -> {
        // if the user has panned away from the location display, turn it on again
        if (locationDisplay.getAutoPanMode() == LocationDisplay.AutoPanMode.OFF) {
          locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
```